### PR TITLE
Openshift minio operator service fix

### PR DESCRIPTION
**Pull request by pjuarezd**
# What this PR do?

When Operator is installed via the Openshift Marketplace, even on a fresh install, operator logs show a error mesage every few seconds, of a client trying to handshake with Operator and failing, this is an example of the error messages:

```
minio-operator-78f89cdd47-jt2nb 2023/01/04 06:41:05 http: TLS handshake error from 10.217.0.35:58566: remote error: tls: bad certificate
minio-operator-78f89cdd47-jt2nb 2023/01/04 06:41:07 http: TLS handshake error from 10.217.0.35:58580: remote error: tls: bad certificate
minio-operator-78f89cdd47-jt2nb 2023/01/04 06:41:20 http: TLS handshake error from 10.217.0.35:36690: remote error: tls: bad certificate
minio-operator-78f89cdd47-jt2nb 2023/01/04 06:41:21 http: TLS handshake error from 10.217.0.35:36694: remote error: tls: bad certificate
minio-operator-78f89cdd47-jt2nb 2023/01/04 06:41:23 http: TLS handshake error from 10.217.0.35:36702: remote error: tls: bad certificate
```

The source of the calls is the kube-apiserver-operator, which is constantly trying to communicate with the Conversion webhook handler that makes the CRD conversion.

We do support CRD conversion, from v1 to v2 of the `tenants.minio.min.io` API, discard the webhook service is not the way to go.

On the kube-apiserver-operator logs messages like the following can be observed:

```
E0104 06:05:37.333718       1 degraded_webhook.go:128] x509: certificate is valid for operator.openshift-operators.svc, operator.openshift-operators.svc.cluster.local, not minio-operator-service.openshift-operators.svc
E0104 06:05:51.710789       1 degraded_webhook.go:128] x509: certificate is valid for operator.openshift-operators.svc, operator.openshift-operators.svc.cluster.local, not minio-operator-service.openshift-operators.svc
E0104 06:05:52.715659       1 degraded_webhook.go:128] x509: certificate is valid for operator.openshift-operators.svc, operator.openshift-operators.svc.cluster.local, not minio-operator-service.openshift-operators.svc
```

Openshift OLM creates an additional service called `minio-operator-service`, there is no option to customize the name of the service, seems like the service is a composition of the operator name + the `service` keyword, ergo `minio-operator-service`. More details about the webhook in the [Defining webhooks](https://docs.openshift.com/container-platform/4.11/operators/operator_sdk/osdk-generating-csvs.html#olm-defining-csv-webhook_osdk-generating-csvs) redhad documentation

This PR adds a aditonal DNSName is to handle the calls from kube-apiserver trough that service, is the easiest way to have a clean operator fresh install, I see no harm on having the addional SAN.

This PR closes https://github.com/minio/operator/issues/1234

After this patch is applied an additional `minio-operator-service.{namespace}.svc` SAN is shown in the certificate
```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            21:6b:96:9a:61:a9:1e:ce:e2:33:b4:5a:27:90:81:aa
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=kubernetes
        Validity
            Not Before: Jan  4 06:28:19 2023 GMT
            Not After : Jan  4 06:28:19 2024 GMT
        Subject: O=system:nodes, CN=system:node:operator.minio-operator.svc
...
            X509v3 Subject Alternative Name: 
                DNS:operator, DNS:operator.minio-operator.svc, DNS:operator.minio-operator.svc.cluster.local, DNS:minio-operator-service.minio-operator.svc
```

This PR also adds the annotation `service.beta.openshift.io/serving-cert-secret-name` to specify to the `kube-apiserver` on which secret the TLS certificate is stored.

This annotation was `alpha` for Openshift v3, and `beta` for openshift v4, based on @kaovilai [post](https://github.com/minio/operator/issues/1234#issuecomment-1267354824)

### CODE REVIEW

File: pkg/controller/cluster/tls.go [hunk 1]
```diff
@@ -298,6 +298,19 @@ func generateCSRCryptoData(serviceName string) ([]byte, []byte, error) {
 		DNSNames:           []string{serviceName, opCommonNoDomain, opCommon},
 	}
 
+	// This sections addresses the edge case "remote error: tls: bad certificate" error in  https://github.com/minio/operator/issues/1234
+	// Openshift OLM creates an additional service `minio-operator-service`, there is no option to choose the name of the service
+	// This aditonal DNSName is to handle the calls from kube-apiserver trough that service, is the easiest way to have a clean operator fresh install.
```
Comment (harshavardhana) [inline pkg/controller/cluster/tls.go:303]:
```suggestion
	// This additional DNSName is to handle the calls from kube-apiserver through that service, which is the easiest way to have a clean operator fresh install.
```

**Review by harshavardhana — DISMISSED** [review]

**Review by kaovilai — APPROVED** [review]
annotations lgtm

**Review by dvaldivia — APPROVED** [review]

**Review by cniackz — APPROVED** [review]
Great fix, thank you very much @pjuarezd for this PR! 👍 
It looks good to me!

**Comment by pjuarezd** [comment]
might be related as well with https://github.com/minio/operator/issues/1331

**Pull request merged**